q2_qemistree/_match.py

```diff
@@ -7,74 +7,65 @@
 # ----------------------------------------------------------------------------
 
 import biom
-import hashlib
 import pandas as pd
 
 
-def match_label(collated_fingerprints: pd.DataFrame,
-                feature_table: biom.Table):
+def match_tables(collated_fingerprints: pd.DataFrame,
+                 feature_table: biom.Table,
+                 feature_data: pd.DataFrame):
     '''
     This function filters the feature table to retain only features with
-    fingerprints. It also relabels features with MD5 hash of its
-    binary fingerprint vector.
+    fingerprints.
 
     Parameters
     ----------
     collated_fingerprints : pd.DataFrame
         table containing mass-spec molecular substructures (columns) for each
         mass-spec feature (index)
     feature_table : biom.Table
-        feature tables with mass-spec feature intensity per sample.
+        feature tables with mass-spec feature intensity per sample
+    feature_data : pd.DataFrame
+        metadata (row ID, row m/z) about features in the feature table
 
     Raises
     ------
     ValueError
         If features in collated fingerprint table are not a subset of
         features in ``feature_table``
+        If feature identifiers in feature table and feature data do not match
+        If 'row m/z' missing in feature data columms
+    UserWarning
+        If features in collated fingerprint table are not a subset of
+        features in corresponding feature table
 
     Returns
     -------
     pd.DataFrame
-        fingerprint table with features relabeled with MD5 hash of
-        its binary fingerprint vector
+        fingerprint table with binary fingerprint vector
     biom.Table
-        feature table that is filtered to contain only the
-        features with predicted fingerprints. Features are labeled by MD5 hash
-        of its binary fingerprint vector
+        feature table that is filtered to contain only the features with
+        predicted fingerprints.
     pd.DataFrame
-        table that maps MD5 hash of a feature to the original feature ID in
-        the input feature table
+        feature data table that is filtered to contain only the features with
+        predicted fingerprints.
     '''
 
     fps = collated_fingerprints.copy()
     allfps = list(fps.index)
-    if fps.empty:
-        raise ValueError("Cannot have empty fingerprint table")
-    table = feature_table.to_dataframe(dense=True)
-    allfeatrs = set(table.index)
+    ftable = feature_table.to_dataframe(dense=True)
+    allfeatrs = set(ftable.index)
+    if 'row m/z' not in feature_data.columns:
+        raise ValueError("Feature data does not contain 'row m/z'")
+    if not set(feature_data.index) == allfeatrs:
```

Comment:
Is this is a safe assumption, are we expecting the feature_data to have exactly the same features as the fingerprints table? I would expect the feature data to have (in general) more features, right? given that not all spectra can produce a fingerprint.

I think we should check that the feature_data is filter to include all the features in `allfps`.